src/awkward/_errors.py

```diff
@@ -43,9 +47,27 @@ def __enter__(self):
             self._slate.__dict__["__primary_context__"] = self
 
     def __exit__(self, exception_type, exception_value, traceback):
-        # Step out of the way so that another ErrorContext can become primary.
-        if self.primary() is self:
-            self._slate.__dict__.clear()
+        try:
+            # Handle caught exception
+            if exception_type is not None:
+                raise self.rewrite_exception(exception_type, exception_value)
+        finally:
+            # Step out of the way so that another ErrorContext can become primary.
+            if self.primary() is self:
+                self._slate.__dict__.clear()
+
+    def rewrite_exception(self, cls: type[E], exception: E) -> E:
+        if issubclass(cls, (NotImplementedError, AssertionError)):
+            # Raise modified exception
+            exc = cls(
+                str(exception)
+                + "\n\nSee if this has been reported at https://github.com/scikit-hep/awkward-1.0/issues"
```

Comment:
Silly comment, but shouldn't you update this URL?

Reply:
Sure, although it redirects. But this one is being shown to users, so it would be less confusing if it were fixed. I'll put in a Suggested Change.